src/tests/suite/dfuse_test.c

```diff
@@ -176,25 +189,154 @@ do_ioctl(void **state)
 	assert_int_equal(dur.uid, geteuid());
 	assert_int_equal(dur.gid, getegid());
 
+	rc = unlinkat(root, "ioctl_file", 0);
+	assert_return_code(rc, errno);
+
 	rc = close(root);
 	assert_return_code(rc, errno);
 }
 
-int
-main(int argc, char **argv)
+static bool
+timespec_gt(struct timespec t1, struct timespec t2)
+{
+	if (t1.tv_sec == t2.tv_sec)
+		return t1.tv_nsec > t2.tv_nsec;
+	else
+		return t1.tv_sec > t2.tv_sec;
+}
+
+void
+do_mtime(void **state)
+{
+	struct stat     stbuf;
+	struct timespec prev_ts;
+	struct timespec now;
+	struct timespec times[2];
+	int             fd;
+	int             rc;
+	char            input_buf[] = "hello";
+	int             root        = open(test_dir, O_PATH | O_DIRECTORY);
+
+	assert_return_code(root, errno);
+
+	/* Open a file and sanity check the mtime */
+	fd = openat(root, "my_file", O_RDWR | O_CREAT | O_EXCL, S_IWUSR | S_IRUSR);
+	assert_return_code(fd, errno);
+	rc = clock_gettime(CLOCK_REALTIME, &now);
+	assert_return_code(fd, errno);
+	rc = fstat(fd, &stbuf);
+	assert_return_code(rc, errno);
+	prev_ts.tv_sec  = stbuf.st_mtim.tv_sec;
+	prev_ts.tv_nsec = stbuf.st_mtim.tv_nsec;
+	assert_true(now.tv_sec - prev_ts.tv_sec < 3);
```

Comment:
should that be
assert_true(timespec_gt(now, stbuf.st_mtim)); ?

Reply:
Yes, I think it probably should.